variables.tf

```diff
@@ -145,3 +145,9 @@ variable "load_balancer_rules_enabled" {
   type        = bool
   default     = false
 }
+
+variable "additional_rules" {
+  description = "Additional network security group rules to add. For arguements please refer to https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/network_security_rule#argument-reference"
+  type        = list(any)
```

Comment:
Use a fully-typed object with `optional` attributes.

Reply:
I don't think it's a good idea as `terraform` 1.3 is still in alpha and the `optional` feature is only experimental in 1.2 and it's not advised to enable it in prod with 1.2.

Reply:
Indeed, that's a good concern !

We've discussed it internally a while ago and we've made the choice to allow the use of the experimental feature in our modules for versions < 1.3.
Also, we will release major versions of our modules for Terraform 1.3+ in case they use the optional implementation.

Reply:
I've used the experimental `optional` 🤞, although I wanted to do it in a separate PR after the official release.

As I've used `default` with `optional`, I've updated the required terraform version to 1.3.

Reply:
Since Terraform 1.3 is not released yet, we will have to go with 1.2 for now.

I will probably make the changes into your branch directly

Reply:
Changes pushed